TwitchPlaysAssembly/Src/Votes.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Timers;
+
+public static class Votes
+{
+	public static bool _going = false;
+	public static bool _global = false;
+	public static Dictionary<string, bool> Voters = new Dictionary<string, bool>();
+	public static TwitchBomb Bomb;
+
+	private static Timer Countdown = null;
+	private static string ActionType = "";
+
+	private static Dictionary<string, Action> actionDict = new Dictionary<string, Action>()
+	{
+		{"Detonation", () => Bomb.CauseExplosionByModuleCommand("Voting ended. Detonating bomb...", "Voted detonation")},
+		{"VS Mode Toggle", () => SendAndDo("Voting ended. Toggling VS Mode...", () => OtherModes.Toggle(TwitchPlaysMode.VS)) }
+	};
+	private static Dictionary<string, int> percentageDict = new Dictionary<string, int>()
+	{
+		{ "Detonation", TwitchPlaySettings.data.MinimumYesForDetonation },
+		{"VS Mode Toggle", TwitchPlaySettings.data.MinimumYesForVSMode }
+	};
+
+	private static void SendAndDo(string message, Action action)
+	{
+		IRCConnection.SendMessage(message);
+		action();
+		return;
+	}
+
+	static Votes()
+	{
+		ResetTimer();
+	}
+
+	#pragma warning disable IDE0049
+	public static void OnTick(Object source, ElapsedEventArgs e)
```

Comment:
I imagine this was supposed to be called `Elapsed` instead of `OnTick` since it's hooked to the `Elapsed` event.

Reply:
I called it OnTick because it gets called when the timer ticks. :D But I can rename it if you think that’s better

Reply:
This is definitely not called when the timer ticks, it's called when the timer expires (i.e. runs out of time): `Countdown.Elapsed += Elapsed;`
So if it's supposed to be (which I don't think it is) then you should fix that.